strawberry/experimental/pydantic/nested/base.py

```diff
@@ -0,0 +1,241 @@
+        # We need to redefine the class each time to make sure
+        # that params passed to typing.Union are of distincts types
+        @dataclass(frozen=True)
+        class LazyForwardRefType(cls):  # type: ignore
+            pass
+
+        model, name, index = params
+        # strawberry.LazyType wants some positional arguments
+        return LazyForwardRefType("", "", None, model, name, index)
+
+    @cached_property
+    def _resolve_type(self) -> Any:
+        # Update type hints to resolve ForwardRefs
+        typ = get_type_hints(self.model)[self.name]
+        _, inner_type = extract_type_list(typ)
+        return inner_type[self.index]._strawberry_type
+
+    @property
+    def _type_definition(self) -> StrawberryType:
+        """Compatibility with strawberry typing system."""
+        return self._resolve_type._type_definition
+
+
+class NestedBackend(ABC):
+    def __init__(self, name: str, model: Type[pydantic.BaseModel]) -> None:
+        self.name = name
+        self.model = model
+        self.child_type: Optional[Union[Tuple[Any, ...], Any]] = None
+        self._post_init()
+        if not self.child_type:
+            raise RelationFieldError()
+
+    @abstractmethod
+    def _post_init(self) -> None:
+        """Called during __init__. Must fill self.child_type if field is a relation."""
+
+    @abstractmethod
+    def get_strawberry_type(self) -> object:
+        """Return the corresponding strawberry type for the field."""
+
+
+class RelationFieldError(Exception):
+    """Field is not a relation (does not reference another type)."""
+
+
+class NestedPydanticBackend(NestedBackend):
+    def _post_init(self):
+        self.field = self.model.__fields__[self.name]
+        if isinstance(self.field.type_, ForwardRef) or (
+            isclass(self.field.type_)
+            and issubclass(self.field.type_, pydantic.BaseModel)
+        ):
+            self.child_type: Tuple[Any, ...] = (self.field.type_,)
+        elif get_origin(self.field.type_) is Union and any(
+            _may_be_model(t) for t in get_args(self.field.type_)
+        ):
+            self.child_type = get_args(self.field.type_)
+
+    def _process_type(self, typ, idx: int = 0) -> object:
+        if isinstance(typ, ForwardRef):
+            return LazyForwardRefType[self.model, self.name, idx]  # type: ignore
+        elif isclass(typ) and issubclass(typ, pydantic.BaseModel):
+            return getattr(typ, "_strawberry_type", LazyModelType[typ])  # type: ignore
+        # Already a strawberry type
+        return typ
+
+    def _process_union(self, union_types: Tuple[Any, ...]) -> Union[Any, Any]:
+        """Build an Union type out of a type params.
+
+        Python < 3.11 don't allow the `typing.Union[*params]` syntax
+        so we build the union iteratively by pairs, using the following property :
+        Union[a, b, c] = Union[Union[a, b], c]
+        """
+        assert len(union_types) > 1
+
+        union: Union[Any, Any] = None
```

Comment:
What is this for? Doesn't this just return `Any`?
```pycon
>>> Union[Any, Any]
typing.Any
```

Reply:
Yeah, I blindly silented mypy I gess. But it doesn't matter anymore since `_process_union` is actually useless.